hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/webapp/FederationBlock.java

```diff
@@ -58,119 +58,195 @@ class FederationBlock extends HtmlBlock {
 
   @Override
   public void render(Block html) {
+
     Configuration conf = this.router.getConfig();
-    boolean isEnabled = conf.getBoolean(
-        YarnConfiguration.FEDERATION_ENABLED,
+    boolean isEnabled = conf.getBoolean(YarnConfiguration.FEDERATION_ENABLED,
         YarnConfiguration.DEFAULT_FEDERATION_ENABLED);
+
     if (isEnabled) {
-      setTitle("Federation");
+
+      List<Map<String, String>> lists = new ArrayList<>();
 
       // Table header
-      TBODY<TABLE<Hamlet>> tbody = html.table("#rms").thead().tr()
+      TBODY<TABLE<Hamlet>> tbody =
+          html.table("#rms").$class("display").$style("width:100%").thead().tr()
           .th(".id", "SubCluster")
-          .th(".submittedA", "Applications Submitted*")
-          .th(".pendingA", "Applications Pending*")
-          .th(".runningA", "Applications Running*")
-          .th(".failedA", "Applications Failed*")
-          .th(".killedA", "Applications Killed*")
-          .th(".completedA", "Applications Completed*")
-          .th(".contAllocated", "Containers Allocated")
-          .th(".contReserved", "Containers Reserved")
-          .th(".contPending", "Containers Pending")
-          .th(".availableM", "Available Memory")
-          .th(".allocatedM", "Allocated Memory")
```

Comment:
Where did all this go?

Reply:
This information is preserved and will be set in `initFederationBlockTableJs`.

For the Federation page, as a user, I first want to know the running status of the SubCluster, and then I care about the capacity information of the cluster. I have made some changes to the information in the displayed list, firstly showing

```
1.subClusterId
2.subClusterName
3.lastStartTime
4.lastHeartBeat
5.resources info
6.nodes info
```

When we click on the cluster id field, the metric information of the cluster will be displayed, which are divided into Application Metric, Resource Metric, and Node Metric. 

When we click on the cluster id field, the metric information of the cluster will be displayed, which are Application Metric, Resource Metric, and Node Metric. When we click on the cluster id again, these information can be hidden, and only the most basic SubCluster status is exposed.